FreeBSD support

It would be nice to have a FreeBSD build, so that we could use Kopia to backup [TrueNAS CORE](https://www.truenas.com/truenas-core/) contents (previously known as FreeNAS).

Also see this short [forum thread](https://kopia.discourse.group/t/compiling-for-freebsd/225) about the same.

@julio-lopez submitted PR https://github.com/kopia/kopia/pull/714 that allowed a FreeBSD build for previous Kopia release, but it's not working anymore for 0.8.0 release.

This is what Im getting whan I try to build a FreeBSD binary on Linux, using Go version 1.16.3, after I applied PR https://github.com/kopia/kopia/pull/714 changes to the 0.8.0 code:
```
$ make -e GOOS=freebsd install-noui
go install -ldflags "-s -w -X github.com/kopia/kopia/repo.BuildVersion=20210411.0.0-50fac71a -X github.com/kopia/kopia/repo.BuildInfo=50fac71a32a26d81418c1df22047e307194ae4e8 -X github.com/kopia/kopia/repo.BuildGitHubRepo="
# github.com/hanwen/go-fuse/v2/fuse
../../../go/pkg/mod/github.com/hanwen/go-fuse/v2@v2.0.4-0.20210104155004-09a3c381714c/fuse/api.go:218:41: undefined: GetAttrIn
../../../go/pkg/mod/github.com/hanwen/go-fuse/v2@v2.0.4-0.20210104155004-09a3c381714c/fuse/api.go:219:41: undefined: SetAttrIn
../../../go/pkg/mod/github.com/hanwen/go-fuse/v2@v2.0.4-0.20210104155004-09a3c381714c/fuse/api.go:222:39: undefined: MknodIn
../../../go/pkg/mod/github.com/hanwen/go-fuse/v2@v2.0.4-0.20210104155004-09a3c381714c/fuse/api.go:246:42: undefined: SetXAttrIn
../../../go/pkg/mod/github.com/hanwen/go-fuse/v2@v2.0.4-0.20210104155004-09a3c381714c/fuse/api.go:252:40: undefined: CreateIn
../../../go/pkg/mod/github.com/hanwen/go-fuse/v2@v2.0.4-0.20210104155004-09a3c381714c/fuse/api.go:254:38: undefined: ReadIn
../../../go/pkg/mod/github.com/hanwen/go-fuse/v2@v2.0.4-0.20210104155004-09a3c381714c/fuse/api.go:263:39: undefined: WriteIn
../../../go/pkg/mod/github.com/hanwen/go-fuse/v2@v2.0.4-0.20210104155004-09a3c381714c/fuse/api.go:272:41: undefined: ReadIn
../../../go/pkg/mod/github.com/hanwen/go-fuse/v2@v2.0.4-0.20210104155004-09a3c381714c/fuse/types.go:550:2: undefined: Attr
../../../go/pkg/mod/github.com/hanwen/go-fuse/v2@v2.0.4-0.20210104155004-09a3c381714c/fuse/types.go:578:2: undefined: Attr
../../../go/pkg/mod/github.com/hanwen/go-fuse/v2@v2.0.4-0.20210104155004-09a3c381714c/fuse/api.go:272:41: too many errors
make: *** [Makefile:51: install-noui] Error 2
```

I guess there are issues with the recent Fuse changes.

Is there any chance for a FreeBSD binary? Anyone else interested in it?

--- a/.goreleaser.yml
+++ b/.goreleaser.yml
@@ -25,6 +25,7 @@ archives:
       darwin: macOS
       linux: linux
       windows: windows
+      freebsd: freebsd-experimental
       amd64: x64
     name_template: "{{ .ProjectName }}-{{ .Version }}-{{ .Os }}-{{ .Arch }}"
     wrap_in_directory: true

--- a/.goreleaser.yml
+++ b/.goreleaser.yml
@@ -8,6 +8,7 @@ builds:
   - CGO_ENABLED=0
   goos:
   - linux
+  - freebsd
   goarch:
     - amd64
     - arm

--- a/internal/mount/mount_fuse.go
+++ b/internal/mount/mount_fuse.go
@@ -1,4 +1,4 @@
-// +build !windows
+// +build !windows,!freebsd
 
 package mount
 

--- a/internal/mount/mount_posix_webdav_helper.go
+++ b/internal/mount/mount_posix_webdav_helper.go
@@ -1,4 +1,4 @@
-// +build !windows
+// +build !windows,!freebsd
 
 package mount
 

--- a/internal/mount/mount_unsupported.go
+++ b/internal/mount/mount_unsupported.go
@@ -1,0 +1,16 @@
+// +build freebsd
+
+package mount
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	"github.com/kopia/kopia/fs"
+)
+
+// Directory returns an error due to mounting being unsupported on current operating system.
+func Directory(ctx context.Context, entry fs.Directory, mountPoint string, mountOptions Options) (Controller, error) {
+	return nil, errors.Errorf("mounting is not supported")
+}

--- a/snapshot/restore/local_fs_output_unix.go
+++ b/snapshot/restore/local_fs_output_unix.go
@@ -1,3 +1,5 @@
+// +build linux freebsd
+
 package restore
 
 import (

